packages/arb-ts/src/lib/l1Bridge.ts

```diff
@@ -89,11 +82,25 @@ export class L1Bridge {
     )
   }
 
-  public async updateAllL1Tokens(): Promise<Tokens> {
-    for (const l1Address in this.l1Tokens) {
-      await this.getAndUpdateL1TokenData(l1Address)
+  public async setSigner(newSigner: Signer) {
+    const newL1Provider = newSigner.provider
+    if (newL1Provider === undefined) {
+      throw new Error('Signer must be connected to an Ethereum provider')
     }
-    return this.l1Tokens
+    // check chainId to ensure its still in the same network.
+    const prevNetwork = await this.l1Provider.getNetwork()
+    const newNetwork = await newL1Provider.getNetwork()
```

Comment:
Let's pipeline these promises

Reply:
Pipelined (also took the opportunity to pipeline in Bridge.init)